test/functional/feature_fee_estimation.py

```diff
@@ -290,6 +291,44 @@ def sanity_check_rbf_estimates(self, utxos):
         est_feerate = node.estimatesmartfee(2)["feerate"]
         assert_equal(est_feerate, high_feerate_kvb)
 
+    def test_old_fee_estimate_file(self):
+        # Get the initial fee rate while node is running
+        fee_rate = self.nodes[0].estimatesmartfee(1)["feerate"]
+
+        # Restart node to ensure fee_estimate.dat file is read
+        self.restart_node(0)
+        assert_equal(self.nodes[0].estimatesmartfee(1)["feerate"], fee_rate)
+
+        fee_dat = fee_dat = self.nodes[0].chain_path / "fee_estimates.dat"
+
+        # Stop the node and backdate the fee_estimates.dat file for 61 hours
+        self.stop_node(0)
+        hour = 60 * 60
+        last_modified_time = time.time() - 61 * hour
+        os.utime(fee_dat, (last_modified_time, last_modified_time))
+
+        # Start node and ensure the fee_estimates.dat file was not read
+        self.start_node(0)
+        assert_equal(self.nodes[0].estimatesmartfee(1)["errors"], ["Insufficient data or no feerate found"])
+
+
+    def test_estimate_dat_is_flushed_periodically(self):
+            fee_dat = self.nodes[0].chain_path / "fee_estimates.dat"
+
+            # Verify that fee_estimates.dat does not exist
+            assert_equal(os.path.isfile(fee_dat), False)
+
+            # Check if the string "Flushed fee estimates to fee_estimates.dat." is present in the debug log file.
+            # If it is, it indicates that fee estimates have been successfully flushed to disk.
+            with self.nodes[0].assert_debug_log(expected_msgs=["Flushed fee estimates to fee_estimates.dat."], timeout=1):
+                # Mock the scheduler for an hour, fee estimates will be flushed to fee_estimate.dat
+                hour = 3600
+                self.nodes[0].mockscheduler(hour)
+
+            # Verify that fee estimates were flushed and fee_estimates.dat file is created
+            assert_equal(os.path.isfile(fee_dat), True)
```

Comment:
might be good to test that it does it *every hour*, and that the output files should be matching provided we don't see blocks in the interim, also, check that what is written on node shutdown also matches?

That way we can have higher assurance on the timing of the flushing and the contents matching what we already do.

Reply:
Sure, thank you, I updated the test.